search-impl/src/main/java/com/example/auction/search/Module.java

```diff
@@ -0,0 +1,30 @@
+package com.example.auction.search;
+
+import com.example.auction.bidding.api.BiddingService;
+import com.example.auction.item.api.ItemService;
+import com.example.auction.search.api.SearchService;
+import com.example.auction.search.impl.BrokerEventConsumer;
+import com.example.auction.search.impl.IndexedStoreImpl;
+import com.example.auction.search.impl.SearchServiceImpl;
+import com.example.elasticsearch.Elasticsearch;
+import com.google.inject.AbstractModule;
+import com.lightbend.lagom.javadsl.server.ServiceGuiceSupport;
+
+/**
+ *
+ */
+public class Module extends AbstractModule implements ServiceGuiceSupport {
+    @Override
+    protected void configure() {
+        bindServices(serviceBinding(SearchService.class, SearchServiceImpl.class));
+
+        bindClient(BiddingService.class);
+        bindClient(ItemService.class);
+        bindClient(Elasticsearch.class);
+
+        bind(IndexedStore.class).to(IndexedStoreImpl.class);
+
+        bind(BrokerEventConsumer.class).asEagerSingleton();
+
```

Comment:
@TimMoore Extracting the topic consumption into a separate class that's how the Module looks like. The `bind(BrokerEventConsumer.class).asEagerSingleton();` is the class that builds the stream and consumes from the broker. It doesn't implement any interface because it doesn't have any method. Everything happens on the constructor so if it can resolve it's dependencies it starts consuming.